src/library_wasi.js

```diff
@@ -206,11 +207,15 @@ var WasiLibrary = {
       if (curr < 0) return -1;
       ret += curr;
       if (curr < len) break; // nothing more to read
+      if (has_offset) {
```

Comment:
Can you make this `if (typeof offset !== 'undefined')` and avoid creating `has_offset` at all?

Reply:
Heh, that was my original version - but I wasn't sure if it was considered OK!